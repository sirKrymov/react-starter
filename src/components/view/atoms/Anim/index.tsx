// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

interface IProps {
  duration?: string;
  children: ReactNode;
  delay?: string;
  type?: 'mount' | 'show';
}

export function Anim({
  children,
  duration = '0.4s',
  delay = '0.2s',
  type = 'show'
}: IProps): ReactElement {
  return (
    <div
      style={{
        animationDuration: duration,
        animationFillMode: 'both',
        animationDelay: delay,
        animationName: type,
        opacity: 0
      }}
    >
      {children}
    </div>
  );
}
