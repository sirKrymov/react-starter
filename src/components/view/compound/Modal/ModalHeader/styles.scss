@import '../../../../../styles/configs';
@import '../../../../../styles/mixins';

.modal-header {
  min-height: 48px;
  padding: 1.5rem 48px 0 1.5rem;

  &__close-btn {
    transition: color $anim-time;
    position: absolute;
    z-index: 10;
    cursor: pointer;
    right: 12px;
    top: 12px;

    &:hover {
      color: var(--textDark);
    }
  }

  &__title {
    margin-bottom: 24px;
  }
}
