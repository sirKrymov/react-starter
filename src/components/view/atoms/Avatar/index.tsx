// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  withBorder?: boolean;
  className?: string;
  onClick?(): void;
  avatar?: string;
  size?: 'xs' | 'sm' | 'lg' | number;
  id: string;
}

export function Avatar({
  className,
  onClick,
  withBorder,
  avatar,
  size = 'xs',
  id
}: IProps): ReactElement {
  const avatarClass = cn(
    {
      avatar: true,
      'avatar--with-border': withBorder,
      'avatar--with-click': onClick,
      ...(typeof size === 'string' && { [`avatar--${size}`]: size })
    },
    className
  );

  return (
    <div
      className={avatarClass}
      onClick={onClick ? onClick : undefined}
      style={
        typeof size === 'number'
          ? {
              height: size,
              width: size
            }
          : {}
      }
      id={id}
    >
      <img
        className="img-cover"
        src={
          avatar ||
          'https://f0.pngfuel.com/png/980/886/male-portrait-avatar-png-clip-art.png'
        }
        alt="avatar"
      />
    </div>
  );
}
