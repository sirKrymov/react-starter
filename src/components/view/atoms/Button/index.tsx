// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { CSSProperties, ReactElement, ReactNode, memo } from 'react';

import cn from 'classnames';

import { Typography } from 'components/view/atoms/Typography';
import { Loader } from 'components/view/atoms/loaders/Loader';
import { UiSize } from 'components/view/atoms/UiSize';

import './styles.scss';

interface IProps {
  fontWeight?: 'default' | 'bold';
  uppercase?: boolean;
  onClick?(): void | undefined;
  className?: string;
  disabled?: boolean;
  children: ReactNode;
  variant?: 'contained' | 'outlined';
  loading?: boolean;
  height?: 'xs' | 'sm' | 'md' | 'lg' | 'full';
  border?: boolean;
  width?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'full' | 'content';
  color?: 'secondary';
  style?: CSSProperties;
  type?: 'button' | 'submit';
  id: string;
}

function ButtonBase({
  fontWeight = 'bold',
  uppercase,
  className,
  disabled,
  children,
  onClick,
  variant = 'contained',
  loading,
  border,
  height,
  width,
  style,
  color = 'secondary',
  type = 'button',
  id
}: IProps): ReactElement {
  const buttonClass = cn(
    {
      button: true,
      [`button--font-weight-${fontWeight}`]: fontWeight,
      [`button--variant-${variant}`]: variant,
      [`button--padding-${height}`]: height && width === 'content',
      [`button--color-${color}`]: color,
      [`button--disabled`]: disabled,
      [`button--border`]: border,
      'button--loading': loading
    },
    className
  );

  return (
    <UiSize width={width} height={height}>
      <button
        className={buttonClass}
        disabled={disabled}
        onClick={loading || disabled ? undefined : onClick}
        style={style}
        type={type}
        id={id}
      >
        {loading ? (
          <Loader color="on-secondary" />
        ) : (
          <div className="button__content">
            <Typography uppercase={uppercase} variant="inherit" color="inherit">
              {children}
            </Typography>
          </div>
        )}
      </button>
    </UiSize>
  );
}

export const Button = memo(ButtonBase);
