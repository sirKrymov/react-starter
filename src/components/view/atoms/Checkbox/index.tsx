// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ChangeEvent, ReactElement } from 'react';

import { IoMdCheckmark } from 'react-icons/io';
import cn from 'classnames';

import './styles.scss';

interface IProps {
  defaultChecked?: boolean;
  onChange(e: ChangeEvent<any>): void;
  className?: string;
  disabled?: boolean;
  value: boolean;
  size?: string;
  name: string;
  id: string;
}

export function Checkbox({
  defaultChecked,
  className,
  onChange,
  disabled,
  value,
  size,
  name,
  id
}: IProps): ReactElement {
  const checkboxClass = cn(
    {
      checkbox: true,
      [`checkbox--size-${size}`]: size,
      'checkbox--disabled': disabled
    },
    className
  );

  return (
    <div className={checkboxClass}>
      <input
        defaultChecked={defaultChecked}
        onChange={onChange}
        disabled={disabled}
        checked={value}
        type="checkbox"
        name={name}
        id={id}
      />

      <label className="checkbox__label" htmlFor={id}>
        <span className="checkbox__icon">
          <IoMdCheckmark />
        </span>
      </label>
    </div>
  );
}
