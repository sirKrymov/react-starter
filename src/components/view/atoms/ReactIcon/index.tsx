// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  onClick?(): void;
  children: ReactNode;
  pointer?: boolean;
  rotate?: '90' | '180' | '270' | '360';
  color?: 'on-primary' | 'secondary' | 'primary' | 'thirdy' | 'valid' | 'error';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'xxxl';
  id?: string;
}

export function ReactIcon({
  className,
  children,
  pointer,
  onClick,
  rotate,
  color = 'primary',
  size = 'lg',
  id
}: IProps): ReactElement {
  const reactIconClass = cn(
    {
      'react-icon': true,
      [`react-icon--rotate-${rotate}`]: rotate,
      [`react-icon--color-${color}`]: color,
      [`react-icon--size-${size}`]: size,
      'react-icon--pointer': pointer
    },
    className
  );

  return (
    <div className={reactIconClass} onClick={onClick} id={id}>
      {children}
    </div>
  );
}
