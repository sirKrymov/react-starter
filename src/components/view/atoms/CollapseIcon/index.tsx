// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { TiArrowSortedDown } from 'react-icons/ti';
import cn from 'classnames';

import { ReactIcon } from 'components/view/atoms/ReactIcon';

import './styles.scss';

interface IProps {
  onClick?(): void;
  open?: boolean;
  id: string;
}

export function CollapseIcon({ onClick, open, id }: IProps): ReactElement {
  const collapseIconClass = cn({
    'collapse-icon': true,
    'collapse-icon--open': open
  });

  return (
    <div className={collapseIconClass} onClick={onClick} id={id}>
      <ReactIcon size="md" color="on-primary">
        <TiArrowSortedDown />
      </ReactIcon>
    </div>
  );
}
