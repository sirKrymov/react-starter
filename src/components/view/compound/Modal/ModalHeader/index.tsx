// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { IoMdCloseCircle } from 'react-icons/io';

import { Typography } from '../../../atoms/Typography';
import { ReactIcon } from '../../../atoms/ReactIcon';

import './styles.scss';

interface IProps {
  onClose?(): void;
  title?: string;
}

export function ModalHeader({ onClose, title }: IProps): ReactElement | null {
  return onClose || title ? (
    <div className="modal-header">
      {title && (
        <Typography className="modal-header__title" variant="h3" noWrap>
          {title}
        </Typography>
      )}

      {onClose && (
        <ReactIcon
          className="modal-header__close-btn"
          onClick={onClose}
          size="xl"
        >
          <IoMdCloseCircle />
        </ReactIcon>
      )}
    </div>
  ) : null;
}
