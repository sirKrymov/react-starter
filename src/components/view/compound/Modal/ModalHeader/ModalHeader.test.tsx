import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import { ModalHeader } from '../ModalHeader';
import { ReactIcon } from '../../../atoms/ReactIcon';

describe('<ModalHeader> compound component', () => {
  const handleClick = sinon.spy();

  let shallowComponent: ShallowWrapper;

  const createComponent = (props?: Record<string, any>) => {
    shallowComponent = shallow(<ModalHeader {...props} />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent({ title: 'Check title', onClose: handleClick });
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });

    it('AND it should handle initial button logic', () => {
      shallowComponent.find(ReactIcon).simulate('click');
      expect(handleClick).toHaveProperty('callCount', 1);
    });
  });

  describe('WHEN the component don`t have title and onClose properties', () => {
    it('THEN it should`t render', () => {
      createComponent();
      expect(shallowComponent.html()).toBe(null);
    });
  });
});
