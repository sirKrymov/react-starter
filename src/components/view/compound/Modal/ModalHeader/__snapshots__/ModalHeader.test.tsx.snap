// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalHeader> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="modal-header"
>
  <Typography
    className="modal-header__title"
    noWrap={true}
    variant="h3"
  >
    Check title
  </Typography>
  <ReactIcon
    className="modal-header__close-btn"
    onClick={[Function]}
    size="xl"
  >
    <IoMdCloseCircle />
  </ReactIcon>
</div>
`;
